clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -69,6 +70,11 @@
     private final boolean autoCommitEnabled;
     private final AutoCommitTask autoCommitTask;
     private final ConsumerInterceptors interceptors;
+    private final boolean excludeInternalTopics;
+    
+    static {
+        INTERNAL_TOPICS.add("__consumer_offsets");
```

Comment:
@ijuma Thats a really good point. The admin scripts have checks for "Internal Topics" [example](https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/admin/TopicCommand.scala#L172) too. As a part of KIP-4 I may need to add a boolean field to the MetadataResponse indicating if a topic is internal so I can support all existing tool functionality. Perhaps this could use that too?